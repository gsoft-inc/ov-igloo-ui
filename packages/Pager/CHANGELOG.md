# @igloo-ui/pager

## 0.1.0

### Minor Changes

- d04afa8: Initial release of Pager component
