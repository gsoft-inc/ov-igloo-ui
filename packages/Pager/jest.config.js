const base = require('../../jest.config');
const { name } = require('./package.json');

module.exports = {
  ...base,
  name,
  displayName: name,
};
