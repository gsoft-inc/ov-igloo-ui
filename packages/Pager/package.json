{
  "name": "@igloo-ui/pager",
  "version": "1.1.0",
  "main": "dist/Pager.js",
  "module": "dist/Pager.js",
  "types": "dist/Pager.d.ts",
  "styles": "dist/pager.css",
  "publishConfig": {
    "access": "public"
  },
  "repository": "git+https://github.com/gsoft-inc/ov-igloo-ui.git",
  "author": "Officevibe Inc.",
  "contributors": [
    "Vicky Comeau vicky.comeau@gsoft.com"
  ],
  "license": "Apache-2.0",
  "peerDependencies": {
    "@igloo-ui/provider": ">=0.0.0",
    "react": ">= 16.8.6"
  },
  "files": [
    "dist"
  ],
  "scripts": {
    "build": "rollup -c rollup.config.js"
  },
  "dependencies": {
    "@igloo-ui/tokens": "^2.1.0",
    "@hopper-ui/tokens": "^4.0.0",
    "classnames": "^2.3.2",
    "@igloo-ui/icons": "^1.12.2"
  },
  "devDependencies": {
    "@igloo-ui/provider": "^0.1.3"
  },
  "browserslist": [
    "> 1%",
    "last 2 versions",
    "IE >= 11"
  ]
}
