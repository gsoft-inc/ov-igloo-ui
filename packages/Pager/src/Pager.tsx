import * as React from 'react';
import cx from 'classnames';
import ChevronLeft from '@igloo-ui/icons/dist/ChevronLeft';
import ChevronRight from '@igloo-ui/icons/dist/ChevronRight';

import { usePagination, JUMPPREV, JUMPNEXT } from './usePagination';
import { EllipsisIcon } from './svgs';

import './pager.scss';

export interface PagerProps extends React.ComponentProps<'div'> {
  /** Add a specific class to the pager component */
  className?: string;
  /** The number of the currently active page */
  currentPage: number;
  /** Add a data-test tag for automated tests */
  dataTest?: string;
  /** Called when the page is changed. */
  onPageChange: (pageNum: number) => void;
  /** Add the number of elements each page should contain */
  pageSize: number;
  /** Represents how many pages should
   * be on each side of the current page.  */
  siblingCount?: number;
  /** The total amount of elements */
  totalCount: number;
}

const Pager: React.FunctionComponent<PagerProps> = (props: PagerProps) => {
  const {
    className,
    currentPage,
    dataTest,
    onPageChange,
    pageSize,
    siblingCount = 1,
    totalCount,
    ...rest
  } = props;

  const paginationRange = usePagination({
    currentPage,
    pageSize,
    siblingCount,
    totalCount,
  });

  const lastPage = paginationRange[paginationRange.length - 1];
  const totalPageCount = Math.ceil(totalCount / pageSize);

  const onNext = (): void => {
    if (totalPageCount > currentPage) {
      onPageChange(currentPage + 1);
    }
  };

  const onPrevious = (): void => {
    if (currentPage > 1) {
      onPageChange(currentPage - 1);
    }
  };

  const onJumpPrevious = (): void => {
    onPageChange(Math.max(currentPage - 5, 1));
  };

  const onJumpNext = (): void => {
    onPageChange(Math.min(currentPage + 5, totalPageCount));
  };

  const renderPrevNext = (type: string): React.ReactNode => {
    const isPrev = type === 'prev';
    const isDisabled = isPrev ? currentPage === 1 : currentPage === lastPage;
    return (
      <li className="ids-pager__item">
        <button
          className={cx('ids-pager__button', `ids-pager__${type}`)}
          onClick={isPrev ? onPrevious : onNext}
          disabled={isDisabled}
          aria-label={isPrev ? 'Go to previous page' : 'Go to next page'}
        >
          {isPrev && <ChevronLeft size="small" />}
          {!isPrev && <ChevronRight size="small" />}
        </button>
      </li>
    );
  };

  return (
    <div className={cx('ids-pager', className)} {...rest}>
      <nav
        aria-label="Pagination"
        className="ids-pager__nav"
        data-test={dataTest}
      >
        <ol className="ids-pager__list">
          {renderPrevNext('prev')}

          {paginationRange.map((pageNumber) => {
            let pagerBtn = (
              <button
                className={cx('ids-pager__button', {
                  'ids-pager__button--selected': pageNumber === currentPage,
                })}
                onClick={() => onPageChange(pageNumber as number)}
                aria-current={pageNumber === currentPage && 'page'}
                aria-label={
                  pageNumber === currentPage
                    ? `Page ${pageNumber}`
                    : `Go to page ${pageNumber}`
                }
              >
                {pageNumber}
              </button>
            );

            if (pageNumber === JUMPPREV) {
              pagerBtn = (
                <button
                  className="ids-pager__button ids-pager__ellipsis"
                  onClick={onJumpPrevious}
                  aria-label="Jump back 5 pages"
                >
                  <EllipsisIcon />
                </button>
              );
            }
            if (pageNumber === JUMPNEXT) {
              pagerBtn = (
                <button
                  className="ids-pager__button ids-pager__ellipsis"
                  onClick={onJumpNext}
                  aria-label="Jump forward 5 pages"
                >
                  <EllipsisIcon />
                </button>
              );
            }

            return (
              <li key={pageNumber.toString()} className="ids-pager__item">
                {pagerBtn}
              </li>
            );
          })}

          {renderPrevNext('next')}
        </ol>
      </nav>
      <div className="ids-pager__results">
        {pageSize * currentPage - pageSize + 1}-{pageSize * currentPage} of{' '}
        {totalCount}
      </div>
    </div>
  );
};

export default Pager;
