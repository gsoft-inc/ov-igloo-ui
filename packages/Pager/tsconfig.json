{
  "extends": "../../tsconfig.build.json",
  "include": ["src/**/*", "../../typings/**/*", "rollup.config.js", "src/intl"],
  "exclude": ["dist"],
  "compilerOptions": {
    "outDir": "./dist",
    "rootDir": "./src",
    "declarationDir": "./dist",
    "composite": true,
    "baseUrl": ".",
    "typeRoots": ["../../typings", "../../node_modules/@types"]
  }
}
