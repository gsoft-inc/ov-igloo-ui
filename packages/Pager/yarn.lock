# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
# yarn lockfile v1


"@igloo-ui/icons@^1.3.0":
  version "1.5.0"
  resolved "https://registry.yarnpkg.com/@igloo-ui/icons/-/icons-1.5.0.tgz#b477678c1b6c8806fc42c5874ce3d8a2a0eb3cb0"
  integrity sha512-nlFlj69/ps3o2N8aEFk2Y3mJ/RZxRtPRQXg7kyeQYwz8m9N7KHU3xYNJT1ZDzDco98XmhViapT+HAFxs9La4xQ==
  dependencies:
    classnames "^2.3.1"
    html-react-parser "^1.2.7"

"@igloo-ui/tokens@^1.0.0":
  version "1.1.1"
  resolved "https://registry.yarnpkg.com/@igloo-ui/tokens/-/tokens-1.1.1.tgz#a5f25912e2f74dec56e845d1845f00c41abecc5a"
  integrity sha512-Nt+5gi7kqzhUuP99V+1INRYDHTa2iTCPUC8AbMK38Iv5zD9+VmokNATwsXjRZA2v3jaiBUgRccpwZgQYNbiBFw==

classnames@^2.3.1:
  version "2.3.2"
  resolved "https://registry.yarnpkg.com/classnames/-/classnames-2.3.2.tgz#351d813bf0137fcc6a76a16b88208d2560a0d924"
  integrity sha512-CSbhY4cFEJRe6/GQzIk5qXZ4Jeg5pcsP7b5peFSDpffpe1cqjASH/n9UTjBwOp6XpMSTwQ8Za2K5V02ueA7Tmw==

dom-serializer@^1.0.1:
  version "1.4.1"
  resolved "https://registry.yarnpkg.com/dom-serializer/-/dom-serializer-1.4.1.tgz#de5d41b1aea290215dc45a6dae8adcf1d32e2d30"
  integrity sha512-VHwB3KfrcOOkelEG2ZOfxqLZdfkil8PtJi4P8N2MMXucZq2yLp75ClViUlOVwyoHEDjYU433Aq+5zWP61+RGag==
  dependencies:
    domelementtype "^2.0.1"
    domhandler "^4.2.0"
    entities "^2.0.0"

domelementtype@^2.0.1, domelementtype@^2.2.0:
  version "2.3.0"
  resolved "https://registry.yarnpkg.com/domelementtype/-/domelementtype-2.3.0.tgz#5c45e8e869952626331d7aab326d01daf65d589d"
  integrity sha512-OLETBj6w0OsagBwdXnPdN0cnMfF9opN69co+7ZrbfPGrdpPVNBUj02spi6B1N7wChLQiPn4CSH/zJvXw56gmHw==

domhandler@4.3.1, domhandler@^4.2.0, domhandler@^4.2.2:
  version "4.3.1"
  resolved "https://registry.yarnpkg.com/domhandler/-/domhandler-4.3.1.tgz#8d792033416f59d68bc03a5aa7b018c1ca89279c"
  integrity sha512-GrwoxYN+uWlzO8uhUXRl0P+kHE4GtVPfYzVLcUxPL7KNdHKj66vvlhiweIHqYYXWlw+T8iLMp42Lm67ghw4WMQ==
  dependencies:
    domelementtype "^2.2.0"

domutils@^2.8.0:
  version "2.8.0"
  resolved "https://registry.yarnpkg.com/domutils/-/domutils-2.8.0.tgz#4437def5db6e2d1f5d6ee859bd95ca7d02048135"
  integrity sha512-w96Cjofp72M5IIhpjgobBimYEfoPjx1Vx0BSX9P30WBdZW2WIKU0T1Bd0kz2eNZ9ikjKgHbEyKx8BB6H1L3h3A==
  dependencies:
    dom-serializer "^1.0.1"
    domelementtype "^2.2.0"
    domhandler "^4.2.0"

entities@^2.0.0:
  version "2.2.0"
  resolved "https://registry.yarnpkg.com/entities/-/entities-2.2.0.tgz#098dc90ebb83d8dffa089d55256b351d34c4da55"
  integrity sha512-p92if5Nz619I0w+akJrLZH0MX0Pb5DX39XOwQTtXSdQQOaYH03S1uIQp4mhOZtAXrxq4ViO67YTiLBo2638o9A==

entities@^3.0.1:
  version "3.0.1"
  resolved "https://registry.yarnpkg.com/entities/-/entities-3.0.1.tgz#2b887ca62585e96db3903482d336c1006c3001d4"
  integrity sha512-WiyBqoomrwMdFG1e0kqvASYfnlb0lp8M5o5Fw2OFq1hNZxxcNk8Ik0Xm7LxzBhuidnZB/UtBqVCgUz3kBOP51Q==

html-dom-parser@1.2.0:
  version "1.2.0"
  resolved "https://registry.yarnpkg.com/html-dom-parser/-/html-dom-parser-1.2.0.tgz#8f689b835982ffbf245eda99730e92b8462c111e"
  integrity sha512-2HIpFMvvffsXHFUFjso0M9LqM+1Lm22BF+Df2ba+7QHJXjk63pWChEnI6YG27eaWqUdfnh5/Vy+OXrNTtepRsg==
  dependencies:
    domhandler "4.3.1"
    htmlparser2 "7.2.0"

html-react-parser@^1.2.7:
  version "1.4.14"
  resolved "https://registry.yarnpkg.com/html-react-parser/-/html-react-parser-1.4.14.tgz#577b7a90be0c61eebbbc488d914ad08398c79ef5"
  integrity sha512-pxhNWGie8Y+DGDpSh8cTa0k3g8PsDcwlfolA+XxYo1AGDeB6e2rdlyv4ptU9bOTiZ2i3fID+6kyqs86MN0FYZQ==
  dependencies:
    domhandler "4.3.1"
    html-dom-parser "1.2.0"
    react-property "2.0.0"
    style-to-js "1.1.1"

htmlparser2@7.2.0:
  version "7.2.0"
  resolved "https://registry.yarnpkg.com/htmlparser2/-/htmlparser2-7.2.0.tgz#8817cdea38bbc324392a90b1990908e81a65f5a5"
  integrity sha512-H7MImA4MS6cw7nbyURtLPO1Tms7C5H602LRETv95z1MxO/7CP7rDVROehUYeYBUYEON94NXXDEPmZuq+hX4sog==
  dependencies:
    domelementtype "^2.0.1"
    domhandler "^4.2.2"
    domutils "^2.8.0"
    entities "^3.0.1"

inline-style-parser@0.1.1:
  version "0.1.1"
  resolved "https://registry.yarnpkg.com/inline-style-parser/-/inline-style-parser-0.1.1.tgz#ec8a3b429274e9c0a1f1c4ffa9453a7fef72cea1"
  integrity sha512-7NXolsK4CAS5+xvdj5OMMbI962hU/wvwoxk+LWR9Ek9bVtyuuYScDN6eS0rUm6TxApFpw7CX1o4uJzcd4AyD3Q==

react-property@2.0.0:
  version "2.0.0"
  resolved "https://registry.yarnpkg.com/react-property/-/react-property-2.0.0.tgz#2156ba9d85fa4741faf1918b38efc1eae3c6a136"
  integrity sha512-kzmNjIgU32mO4mmH5+iUyrqlpFQhF8K2k7eZ4fdLSOPFrD1XgEuSBv9LDEgxRXTMBqMd8ppT0x6TIzqE5pdGdw==

style-to-js@1.1.1:
  version "1.1.1"
  resolved "https://registry.yarnpkg.com/style-to-js/-/style-to-js-1.1.1.tgz#417786986cda61d4525c80aed9d1123a6a7af9b8"
  integrity sha512-RJ18Z9t2B02sYhZtfWKQq5uplVctgvjTfLWT7+Eb1zjUjIrWzX5SdlkwLGQozrqarTmEzJJ/YmdNJCUNI47elg==
  dependencies:
    style-to-object "0.3.0"

style-to-object@0.3.0:
  version "0.3.0"
  resolved "https://registry.yarnpkg.com/style-to-object/-/style-to-object-0.3.0.tgz#b1b790d205991cc783801967214979ee19a76e46"
  integrity sha512-CzFnRRXhzWIdItT3OmF8SQfWyahHhjq3HwcMNCNLn+N7klOOqPjMeG/4JSu77D7ypZdGvSzvkrbyeTMizz2VrA==
  dependencies:
    inline-style-parser "0.1.1"
