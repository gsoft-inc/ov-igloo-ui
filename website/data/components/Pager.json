[
  [
    {
      "tags": {},
      "filePath": "/Users/franck.gaudin/Devel/Igloo/components/packages/Pager/src/Pager.tsx",
      "description": "",
      "displayName": "Pager",
      "methods": [],
      "props": {
        "className": {
          "defaultValue": null,
          "description": "Add a specific class to the pager component",
          "name": "className",
          "parent": {
            "fileName": "components/packages/Pager/src/Pager.tsx",
            "name": "PagerProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Pager/src/Pager.tsx",
              "name": "PagerProps"
            }
          ],
          "required": false,
          "type": { "name": "string" }
        },
        "currentPage": {
          "defaultValue": null,
          "description": "The number of the currently active page",
          "name": "currentPage",
          "parent": {
            "fileName": "components/packages/Pager/src/Pager.tsx",
            "name": "PagerProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Pager/src/Pager.tsx",
              "name": "PagerProps"
            }
          ],
          "required": true,
          "type": { "name": "number" }
        },
        "dataTest": {
          "defaultValue": null,
          "description": "Add a data-test tag for automated tests",
          "name": "dataTest",
          "parent": {
            "fileName": "components/packages/Pager/src/Pager.tsx",
            "name": "PagerProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Pager/src/Pager.tsx",
              "name": "PagerProps"
            }
          ],
          "required": false,
          "type": { "name": "string" }
        },
        "onPageChange": {
          "defaultValue": null,
          "description": "Called when the page is changed.",
          "name": "onPageChange",
          "parent": {
            "fileName": "components/packages/Pager/src/Pager.tsx",
            "name": "PagerProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Pager/src/Pager.tsx",
              "name": "PagerProps"
            }
          ],
          "required": true,
          "type": { "name": "(pageNum: number) => void" }
        },
        "pageSize": {
          "defaultValue": null,
          "description": "Specify the number of elements each page should contain",
          "name": "pageSize",
          "parent": {
            "fileName": "components/packages/Pager/src/Pager.tsx",
            "name": "PagerProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Pager/src/Pager.tsx",
              "name": "PagerProps"
            }
          ],
          "required": true,
          "type": { "name": "number" }
        },
        "siblingCount": {
          "defaultValue": null,
          "description": "Represents how many pages should\nbe on each side of the current page.",
          "name": "siblingCount",
          "parent": {
            "fileName": "components/packages/Pager/src/Pager.tsx",
            "name": "PagerProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Pager/src/Pager.tsx",
              "name": "PagerProps"
            }
          ],
          "required": false,
          "type": { "name": "number" }
        },
        "totalCount": {
          "defaultValue": null,
          "description": "The total amount of elements",
          "name": "totalCount",
          "parent": {
            "fileName": "components/packages/Pager/src/Pager.tsx",
            "name": "PagerProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Pager/src/Pager.tsx",
              "name": "PagerProps"
            }
          ],
          "required": true,
          "type": { "name": "number" }
        }
      }
    }
  ],
  [
    {
      "tags": {},
      "filePath": "/Users/franck.gaudin/Devel/Igloo/components/packages/Pager/src/usePagination.tsx",
      "description": "",
      "displayName": "usePagination",
      "methods": [],
      "props": {
        "currentPage": {
          "defaultValue": null,
          "description": "The number of the currently active page",
          "name": "currentPage",
          "parent": {
            "fileName": "components/packages/Pager/src/usePagination.tsx",
            "name": "PaginationProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Pager/src/usePagination.tsx",
              "name": "PaginationProps"
            }
          ],
          "required": true,
          "type": { "name": "number" }
        },
        "pageSize": {
          "defaultValue": null,
          "description": "The number of elements each page should contain",
          "name": "pageSize",
          "parent": {
            "fileName": "components/packages/Pager/src/usePagination.tsx",
            "name": "PaginationProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Pager/src/usePagination.tsx",
              "name": "PaginationProps"
            }
          ],
          "required": true,
          "type": { "name": "number" }
        },
        "siblingCount": {
          "defaultValue": null,
          "description": "Represents how many pages should\nbe on each side of the current page.",
          "name": "siblingCount",
          "parent": {
            "fileName": "components/packages/Pager/src/usePagination.tsx",
            "name": "PaginationProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Pager/src/usePagination.tsx",
              "name": "PaginationProps"
            }
          ],
          "required": true,
          "type": { "name": "number" }
        },
        "totalCount": {
          "defaultValue": null,
          "description": "The total amount of elements",
          "name": "totalCount",
          "parent": {
            "fileName": "components/packages/Pager/src/usePagination.tsx",
            "name": "PaginationProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Pager/src/usePagination.tsx",
              "name": "PaginationProps"
            }
          ],
          "required": true,
          "type": { "name": "number" }
        }
      }
    }
  ]
]
