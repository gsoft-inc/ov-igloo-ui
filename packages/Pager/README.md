# Pager

Pagers allow users to navigate through items of the same table or list.

<Example is="custom" />
<ReferenceLinks is="custom" />

## Installation

To install `@igloo-ui/pager` in your project, you will need to run the following command using [npm](https://www.npmjs.com/):

```bash
npm install @igloo-ui/pager
```

If you prefer [Yarn](https://classic.yarnpkg.com/en/), use the following command instead:

```bash
yarn add @igloo-ui/pager
```

## Usage

Then to use the component in your code, import it.
It also requires the Provider package from Igloo to receive the correct locale.
This should be wrapped around the entire application.

```jsx
import Pager from '@igloo-ui/pager';

const [currentPage, setCurrentPage] = React.useState(1);

  <Pager
    pageSize={5}
    totalCount={200}
    currentPage={currentPage}
    onPageChange={(page) => setCurrentPage(page)}
  />
```


## Internationalization

The Pager component uses the [`@igloo-ui/provider`](https://igloo.officevibe.design/component/Provider) package to receive the correct locale. This should be wrapped around the entire application.

```jsx
import IglooProvider from '@igloo-ui/provider';

<IglooProvider locale="fr-CA">
  <Pager
    pageSize={5}
    totalCount={200}
    currentPage={currentPage}
    onPageChange={(page) => setCurrentPage(page)}
  />
</IglooProvider>
```
