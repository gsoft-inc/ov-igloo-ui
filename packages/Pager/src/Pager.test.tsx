/**
 * @jest-environment jsdom
 */
import React from 'react';
import { render, screen } from '@testing-library/react';

import Pager from './Pager';

describe('Pager', () => {
  let currentPage = 1;
  const handlePageChange = (page: number) => {
    currentPage = page;
  };

  test('It should render without error and a snapshot', () => {
    const { asFragment } = render(
      <Pager
        dataTest="pager1"
        pageSize={5}
        totalCount={50}
        currentPage={currentPage}
        onPageChange={handlePageChange}
      />
    );
    expect(screen.getByTestId('pager1')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  test('It should render a disabled left chevron when on page 1', () => {
    const { container } = render(
      <Pager
        dataTest="pager2"
        pageSize={5}
        totalCount={50}
        currentPage={currentPage}
        onPageChange={handlePageChange}
      />
    );
    const leftChevron = container.querySelector('.ids-pager__prev');
    expect(leftChevron).toBeInTheDocument();
    expect(leftChevron).toBeDisabled();
  });

  test('Current page should be 2 with the appropriate aria tags', () => {
    let currentPage = 2;
    const { container } = render(
      <Pager
        dataTest="pager3"
        pageSize={5}
        totalCount={50}
        currentPage={currentPage}
        onPageChange={handlePageChange}
      />
    );
    const selectedPage = container.querySelector(
      '.ids-pager__button--selected'
    );
    expect(selectedPage).toHaveTextContent('2');
    expect(selectedPage).toHaveAttribute('aria-current', 'page');
    expect(selectedPage).toHaveAttribute('aria-label', 'Page 2');
  });

  test('Buttons should contain appropriate text', () => {
    let currentPage = 5;
    const { container } = render(
      <Pager
        dataTest="pager4"
        pageSize={5}
        totalCount={50}
        currentPage={currentPage}
        onPageChange={handlePageChange}
      />
    );
    const page1 = container.querySelector('li:nth-child(2) .ids-pager__button');
    const ellipsis1Btn = container.querySelector(
      'li:nth-child(3) .ids-pager__button'
    );
    const page4 = container.querySelector('li:nth-child(4) .ids-pager__button');
    const page5 = container.querySelector('li:nth-child(5) .ids-pager__button');
    const page6 = container.querySelector('li:nth-child(6) .ids-pager__button');
    const ellipsis2Btn = container.querySelector(
      'li:nth-child(7) .ids-pager__button'
    );
    const page10 = container.querySelector(
      'li:nth-child(8) .ids-pager__button'
    );
    const ellipsis = container.querySelector(
      'li:nth-child(3) .ids-pager__button svg'
    ) as SVGElement;
    const ellipsis2 = container.querySelector(
      'li:nth-child(7) .ids-pager__button svg'
    ) as SVGElement;
    expect(page1).toHaveTextContent('1');
    expect(ellipsis1Btn).toContainElement(ellipsis);
    expect(page4).toHaveTextContent('4');
    expect(page5).toHaveTextContent('5');
    expect(page6).toHaveTextContent('6');
    expect(ellipsis2Btn).toContainElement(ellipsis2);
    expect(page10).toHaveTextContent('10');
  });
});
