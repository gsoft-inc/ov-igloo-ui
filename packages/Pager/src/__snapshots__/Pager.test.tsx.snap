// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pager It should render without error and a snapshot 1`] = `
<DocumentFragment>
  <div
    class="ids-pager"
  >
    <nav
      aria-label="Pagination"
      class="ids-pager__nav"
      data-test="pager1"
    >
      <ol
        class="ids-pager__list"
      >
        <li
          class="ids-pager__item"
        >
          <button
            aria-label="Go to previous page"
            class="ids-pager__button ids-pager__prev"
            disabled=""
          >
            <svg
              class="ids-icon"
              fill="none"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.333 4 5.57 7.764a.333.333 0 0 0 0 .472L9.333 12"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.5"
              />
            </svg>
          </button>
        </li>
        <li
          class="ids-pager__item"
        >
          <button
            aria-current="page"
            aria-label="Page 1"
            class="ids-pager__button ids-pager__button--number ids-pager__button--selected"
          >
            1
          </button>
        </li>
        <li
          class="ids-pager__item"
        >
          <button
            aria-current="false"
            aria-label="Go to page 2"
            class="ids-pager__button ids-pager__button--number"
          >
            2
          </button>
        </li>
        <li
          class="ids-pager__item"
        >
          <button
            aria-current="false"
            aria-label="Go to page 3"
            class="ids-pager__button ids-pager__button--number"
          >
            3
          </button>
        </li>
        <li
          class="ids-pager__item"
        >
          <button
            aria-current="false"
            aria-label="Go to page 4"
            class="ids-pager__button ids-pager__button--number"
          >
            4
          </button>
        </li>
        <li
          class="ids-pager__item"
        >
          <button
            aria-current="false"
            aria-label="Go to page 5"
            class="ids-pager__button ids-pager__button--number"
          >
            5
          </button>
        </li>
        <li
          class="ids-pager__item"
        >
          <button
            aria-label="Jump forward 5 pages"
            class="ids-pager__button ids-pager__ellipsis"
          >
            <svg
              fill="none"
              height="12"
              viewBox="0 0 12 12"
              width="12"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6 7C6.55228 7 7 6.55228 7 6C7 5.44772 6.55228 5 6 5C5.44772 5 5 5.44772 5 6C5 6.55228 5.44772 7 6 7Z"
                fill="currentColor"
              />
              <path
                d="M2 7C2.55228 7 3 6.55228 3 6C3 5.44772 2.55228 5 2 5C1.44772 5 1 5.44772 1 6C1 6.55228 1.44772 7 2 7Z"
                fill="currentColor"
              />
              <path
                d="M10 7C10.5523 7 11 6.55228 11 6C11 5.44772 10.5523 5 10 5C9.44772 5 9 5.44772 9 6C9 6.55228 9.44772 7 10 7Z"
                fill="currentColor"
              />
            </svg>
          </button>
        </li>
        <li
          class="ids-pager__item"
        >
          <button
            aria-current="false"
            aria-label="Go to page 10"
            class="ids-pager__button ids-pager__button--number"
          >
            10
          </button>
        </li>
        <li
          class="ids-pager__item"
        >
          <button
            aria-label="Go to next page"
            class="ids-pager__button ids-pager__next"
          >
            <svg
              class="ids-icon"
              fill="none"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m6 4 3.823 3.823a.25.25 0 0 1 0 .354L6 12"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.5"
              />
            </svg>
          </button>
        </li>
      </ol>
    </nav>
    <div
      class="ids-pager__results"
    >
      1-5 
of 50
    </div>
  </div>
</DocumentFragment>
`;
